### PR TITLE
Refactoring templates

### PR DESCRIPTION
**Pull request by alexblnn**
Change template structure and add beta template. Addresses issues #14 and #15 

Side note : many functions/variables names remain unclear (as pointed out in #18). This is not addressed here but will be in an upcoming PR.

### CODE REVIEW

File: sanssouci/lambda_calibration.py [hunk 1]
```diff
@@ -169,7 +169,8 @@ def get_pivotal_stats(p0, t_inv=t_inv_linear, K=-1):
     p0 = np.sort(p0, axis=1)
 
     # Step 3: apply template function
-    tkInv_all = t_inv(p0)
+    B, p = p0.shape
+    tkInv_all = np.array([t_inv(p0[:, i], i + 1, p) for i in range(p)]).T
```
Comment (bthirion) [inline sanssouci/lambda_calibration.py:173]: Can you explain more in detail what is done here ?

File: sanssouci/reference_families.py [hunk 2]
```diff
@@ -1,66 +1,87 @@
...
-
-
     Parameters
     ----------
-
-    p0 : array-like of shape (B, p)
-        A numpy array of size [B,p] of null p-values obtained from
-        B permutations for p hypotheses.
-
+    y : array of floats
```
Comment (bthirion) [inline sanssouci/reference_families.py:14]: Give the array shape

File: sanssouci/reference_families.py [hunk 3]
```diff
@@ -1,66 +1,87 @@
...
+    return beta.ppf(alpha, np.arange(1, k + 1),
+                    np.array([m + 1] * k) - np.arange(1, k + 1))
+
 
-    array-like of shape (K,) :
+def t_inv_beta(y, k, m):
+    """
+    Parameters
+    ----------
+    y : array of floats
```
Comment (bthirion) [inline sanssouci/reference_families.py:75]: Shape...

File: sanssouci/reference_families.py [hunk 4]
```diff
@@ -1,66 +1,87 @@
...
+        values to apply template to
+    k : int
+        index of rejection set in beta template
+    m : int
+        number of hypotheses
```
Comment (bthirion) [inline sanssouci/reference_families.py:80]: drop a line here

**Comment by alexblnn** [comment]
As pointed out earlier today, there is potentially a bug on the implementation of beta templates. I will investigate this tomorrow; let's wait before then to merge

**Comment by pneuvial** [comment]
> As pointed out earlier today, there is potentially a bug on the implementation of beta templates. I will investigate this tomorrow; let's wait before then to merge

Note: The bug was not here, so we are all good.

**Comment by pneuvial** [comment]
Great work, thanks!

**Pull request merged**